emily/handler/src/api/handlers/deposit.rs

```diff
@@ -213,6 +213,21 @@ pub async fn create_deposit(
         context: EmilyContext,
         body: CreateDepositRequestBody,
     ) -> Result<impl warp::reply::Reply, Error> {
+        // Reject if we already have a deposit with the same txid and output index and it is NOT pending or reprocessing.
+        let entry = accessors::get_deposit_entry(
+            &context,
+            &DepositEntryKey {
+                bitcoin_txid: body.bitcoin_txid.clone(),
+                bitcoin_tx_output_index: body.bitcoin_tx_output_index,
+            },
+        )
+        .await;
```

Comment:
As `get_deposit_entry` can fail for different reasons, we should propagate the error if it's not `Error::NotFound`, otherwise in case of unrelated errors (eg, dynamodb throttling) we may skip the check.